File: app/src/test/java/com/applify/securepass/crypto/TestCrypto.java
package com.applify.securepass.crypto;

import org.junit.Test;
import java.security.SecureRandom;
import java.util.Base64;
import static org.junit.Assert.*;

public class TestCrypto {
    @Test
    public void testKeyDerivation() {
        try {
            byte[] salt = new byte[16];
            new SecureRandom().nextBytes(salt);
            System.out.println("Salt (Base64): " + Base64.getEncoder().encodeToString(salt));

            String testPin = "123456";
            var key = CryptoManager.deriveKey(testPin, salt);

            System.out.println("Key algorithm: " + key.getAlgorithm());
            System.out.println("Key length: " + key.getEncoded().length * 8 + " bits");

            var key2 = CryptoManager.deriveKey(testPin, salt);
            boolean same = java.util.Arrays.equals(key.getEncoded(), key2.getEncoded());
            System.out.println("Same key? " + same);
            
            assertTrue("Derived keys should be the same for the same input", same);

        } catch (Exception e) {
            e.printStackTrace();
            fail(e.getMessage());
        }
    }
}

File: app/src/test/java/com/applify/securepass/crypto/TestCrypto.kt
package com.applify.securepass.crypto

import org.junit.Assert.assertTrue
import org.junit.Assert.fail
import org.junit.Test
import java.security.SecureRandom
import java.util.Arrays
import java.util.Base64

class TestCrypto {
    @Test
    fun testKeyDerivation() {
        try {
            val salt = ByteArray(16)
            SecureRandom().nextBytes(salt)
            println("Salt (Base64): " + Base64.getEncoder().encodeToString(salt))

            val testPin = "123456"
            val key = CryptoManager.deriveKey(testPin, salt)

            println("Key algorithm: " + key.algorithm)
            println("Key length: " + key.encoded.size * 8 + " bits")

            val key2 = CryptoManager.deriveKey(testPin, salt)
            val same = Arrays.equals(key.encoded, key2.encoded)
            println("Same key? $same")

            assertTrue("Derived keys should be the same for the same input", same)

        } catch (e: Exception) {
            e.printStackTrace()
            fail(e.message)
        }
    }
}